common/datavalues/src/types/type_timestamp.rs

```diff
@@ -56,21 +73,21 @@ impl TimestampType {
 
     #[inline]
     pub fn utc_timestamp(&self, v: i64) -> DateTime<Utc> {
-        let v = v * 10_i64.pow(9 - self.precision as u32);
+        let v = v * 10_i64.pow(6 - self.precision as u32);
```

Comment:
Where did the check `self.precision <= 6`?

Reply:
I think we should check it in datatype, it's datatype limit not sql.

Reply:
yes, we only register timestamp datatype with precision from 0 to 6
https://github.com/Veeupup/databend/blob/a4c2671481/common/datavalues/src/types/type_factory.rs#L53-L63